### ISSUE TITLE
Ensure web components follow accessibility standards

### ISSUE THREAD

**Issue by jnurthen** [adobe/spectrum-web-components#292]
### Expected Behaviour

Web Components should meet WCAG 2.1 AA https://www.w3.org/TR/WCAG21/
Behaviours of components should follow ARIA Authoring Practices https://www.w3.org/TR/wai-aria-practices/

### Actual Behaviour
Many components do not appear to have accessibility behaviours defined correctly. Spot checking 1 or 2 components I found many issues
For example
- [x] action-menu. Menuitems have button child elements. This is not allowed as it leads to an incorrect accessibility tree
- [ ] textfield and other input elements. I don't see any way to label a textfield
- [x] tab doesn't appear to have any roles defined. It also has no visible focus indicator and does not follow the ARIA authoring practices.

### Reproduce Scenario (including but not limited to)
View the above components on the documentation site.

#### Steps to Reproduce
Use Accessibility inspection tools to verify the behaviour of the components.

**Comment by adixon-adobe** [comment]
@Westbrook did you ever get unittests with `axecore` working?

**Comment by Westbrook** [comment]
They work...I've just not added them as some of the Spectrum settings fail. All new components get them by default in the templates added in #591 a la:
```
    it('loads default {{name}} accessibly', async () => {
        const el = await fixture<{{className name}}>(
            html`
                <{{> tagnamePartial}}></{{> tagnamePartial}}>
                `
            );
        
        await elementUpdated(el);

        expect(el).to.be.accessible();
    });
```
So using that pattern across other elements and taking responsibility for keeping it/removing it based on the results is a great idea!

**Comment by Westbrook** [comment]
With the addition of #905 the only task explicitly outlined in this ticket is duplicated in #742.

Closing as duplicate.
